browser/src/app/Log.ts

```diff
@@ -0,0 +1,109 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Logger contains methods for logging the activity
+ */
+
+type Direction = 'INCOMING' | 'OUTGOING';
+interface LogMsg {
+	msg: string;
+	direction: Direction;
+	time: number;
+}
+
+class Logger {
+	private _logs: LogMsg[];
+	private startTime: number;
+
+	constructor() {
+		this._logs = [];
+		this.startTime = null;
+	}
+
+	public log(msg: string, direction: Direction): void {
+		var time = Date.now();
```

Comment:
here and below, let's use `let` instead of `var`.

Also maybe new line after the `if ()` or `while ()`. This would make things more readable.

Reply:
:-(